cmd/storage-rest-server.go

```diff
@@ -47,6 +48,23 @@ func (s *storageRESTServer) writeErrorResponse(w http.ResponseWriter, err error)
 	w.(http.Flusher).Flush()
 }
 
+type bulkErrorsResponse struct {
```

Comment:
this should be an exported struct @vadmeste and also we should have `Errs []error` contextual such as `Errs []error{Name: path, Err: err}`

Reply:
> this should be an exported struct @vadmeste and also we should have `Errs []error` contextual such as `Errs []error{Name: path, Err: err}`

What's the purpose of doing contextual error here ? Multiple Delete Objects already provide such detailed information such as:

```
[REQUEST objectAPIHandlers.DeleteMultipleObjectsHandler-fm] [155724812.829024] [2019-05-07 17:55:28 +0100]                                                                       POST /testbucket/?delete=                                                                                                                                                        Host: localhost:9000                                                                                                                                                             X-Amz-Date: 20190507T165528Z                                                                                                                                                     User-Agent: MinIO (linux; amd64) minio-go/v6.0.25                                                                                         
Content-Length: 846                                                                                                                                                              
Authorization: AWS4-HMAC-SHA256 Credential=minio/20190507/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=c7a110ebed51fd0c4e
ca6da2957243a1660f4ea01af25a954cb77427b531112f                                                                                                                                   
Content-Md5: Vq0MMWAGP6BksPA8wqzUrw==                                                                                                                                            
X-Amz-Content-Sha256: 84f3e1fcb7c2200ed733b958b843acc39519544c1a27f4059767b8ac5dc10588                                                    
                                                                                                                                                                                 
<Delete><Quiet>true</Quiet><Object><Key>testobject-0</Key></Object><Object><Key>testobject-1</Key></Object><Object><Key>testobject-10</Key></Object><Object><Key>testobject-11</K
ey></Object><Object><Key>testobject-12</Key></Object><Object><Key>testobject-13</Key></Object><Object><Key>testobject-14</Key></Object><Object><Key>testobject-15</Key></Object><
Object><Key>testobject-16</Key></Object><Object><Key>testobject-17</Key></Object><Object><Key>testobject-18</Key></Object><Object><Key>testobject-19</Key></Object><Object><Key>t
estobject-2</Key></Object><Object><Key>testobject-3</Key></Object><Object><Key>testobject-4</Key></Object><Object><Key>testobject-5</Key></Object><Object><Key>testobject-6</Key>
</Object><Object><Key>testobject-7</Key></Object><Object><Key>testobject-8</Key></Object><Object><Key>testobject-9</Key></Object></Delete>
                                                         
[RESPONSE] [155724812.829024] [2019-05-07 17:55:28 +0100]
200 OK                            
X-Amz-Request-Id: 159C74EC09D7E285                         
X-Minio-Deployment-Id: 9170a827-e9c0-44ae-bfda-63063b8b4b21
Server: MinIO/DEVELOPMENT.2019-05-07T16-55-17Z
Accept-Ranges: bytes
Content-Type: application/xml
Vary: Origin
X-Xss-Protection: 1; mode=block
Content-Security-Policy: block-all-mixed-content

<?xml version="1.0" encoding="UTF-8"?>
<DeleteResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Error><Code>InternalError</Code><Message>We encountered an internal error, please try again.</Message><Key>testobj
ect-10</Key></Error><Error><Code>InternalError</Code><Message>We encountered an internal error, please try again.</Message><Key>testobject-15</Key></Error></DeleteResult>

```

Reply:
The reasoning is you can avoid doing the translation code such as toName() etc. Also, the APIs of DeleteFileBulk and DeleteObjects will start looking similar - that is the intention of this PR and not repeat the work again.